flybywiresim/aircraft
# feat(GSX): Automatically start Refuel Process & allow Instant Refuel with GSX Fuel Sync enabled

Fixes #8406 

## Summary of Changes

- FBW Refuel Process is started automatically when GSX Fuel-Hose is connected
- Instant-Refuel is still possible with GSX Fuel Sync enabled
- EFB Fuel Page Reflects State of the GSX Refuel Service (and indicates more clearer that the User-Interaction is locked)

## Screenshots (if necessary)


GSX Fuel Sync enabled, Real (or Fast) Refuel is selected:
![image](https://github.com/flybywiresim/aircraft/assets/78614902/26e4fba2-8405-40ed-8827-4daa11c9d6b1)

GSX Refuel was called (Truck on its way to Fuel-Hose Connected)
![image](https://github.com/flybywiresim/aircraft/assets/78614902/dedc9eb7-26c6-41df-9d92-745ddc76060a)

Refueling is active
![image](https://github.com/flybywiresim/aircraft/assets/78614902/7f573f51-4061-4cfb-a213-f96202949d5c)


Instant Refuel allowed while GSX Fuel Sync is enabled in the Settings
![instant-refuel-with-sync-enabled](https://github.com/flybywiresim/aircraft/assets/78614902/7e145412-52af-4331-a1b9-75c5f168d74a)


## References





## Additional context

The Motivation is to remove manual Steps in the EFB when used in Conjunction with GSX. A User requesting Refuel in the GSX Menu already formulated/stated the Intention to Refuel the Plane. So an additional Confirmation is not really necessary.
On Top of that these Kind of Automation is a Preliminary to solve Issues like #7772.
With that Users using external GSX Service Automations (like included in my PilotsDeck/StreamDeck Profile for the FBW) would need less manual Steps in the EFB to enjoy these Automations.

I'm aware that this Change would also start the Refuel-Process even when the User did not make a single Interaction on the EFB (even not powered on). In that Case the FBW Default of 9524kg desired Fuel would be applied. But please consider:
- The User is still able to correct such a Situation with Instant-Refuel (because that is now possible with GSX Fuel Sync enabled)
- The User is still able to set the correct desired Fuel and then call GSX Refuel again, if so desired
- Looking into Auto-Importing the SimBrief Data to Fuel & Payload would be my next Target
- It is not exactly a "best Practice" Flight-Preparation to call Refuel straight away and then don't care about the required Fuel 😅 




Discord username (if different from GitHub): Fragtality

## Testing instructions


1. Load into the Plane
2. Ensure GSX Fuel Sync is enabled in the Settings>3rd Party
3. Do your normal Preparation Flow, but ensure you set/import your desired Fuel on the Ground>Fuel Page in the EFB.
4. Call Refuel in GSX Menu => No further manual Interaction should be needed for the Refuel Process to start. Also: Start Button should not be visible with GSX Fuel Sync on and Fast/Real selected.
5. After the Fuel Truck leaves, change the desired Fuel in the EFB and switch to Instant. Press Play => Instant (De)Fuel should be applied and the Button should be there and clickable.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
